File: LeetCode/cpp/739.cpp
class Solution {
public:
    vector<int> dailyTemperatures(vector<int> &T) {
        int n = T.size();
        stack<int> s;
        vector<int> ret(n, 0);
        for (int i = 0; i < n; ++i) {
            while (!s.empty() && T[s.top()] < T[i]) {
                ret[s.top()] = i - s.top();
                s.pop();
            }
            s.push(i);
        }
        return ret;
    }
};

File: LeetCode/python3/739.py
class Solution:
    def dailyTemperatures(self, T: List[int]) -> List[int]:
        ret = [0 for i in range(len(T))]
        stack = []
        for i in range(len(T) - 1, -1, -1):
            while stack and T[i] >= T[stack[-1]]:
                stack.pop()
            if stack:
                ret[i] = stack[-1] - i
            stack.append(i)
        return ret
